pkg/doc/signature/ed25519signature2018/suite.go

```diff
@@ -36,6 +42,89 @@ type signer interface {
 	Sign(data []byte) ([]byte, error)
 }
 
+// CryptoSigner defines a signer which uses crypto instance to sing the message using particular key handle.
+type CryptoSigner struct {
+	cr crypto.Crypto
+	kh interface{}
+}
+
+// NewCryptoSigner creates a new CryptoSigner.
+func NewCryptoSigner(cr crypto.Crypto, kh interface{}) *CryptoSigner {
+	return &CryptoSigner{
+		cr: cr,
+		kh: kh,
+	}
+}
+
+// Sign will sign document and return signature
+func (s CryptoSigner) Sign(msg []byte) ([]byte, error) {
+	return s.cr.Sign(msg, s.kh)
+}
+
+type verifier interface {
+	// Verify will verify a signature.
+	Verify(pubKeyValue interface{}, doc, signature []byte) error
+}
+
+// LocalCryptoVerifier defines a verifier which is based on Local KMS and Crypto
+// which uses keyset.Handle as input for verification.
+type LocalCryptoVerifier struct {
+	cr       crypto.Crypto
+	localKMS *localkms.LocalKMS
+}
+
+// NewLocalCryptoVerifier creates a new LocalCryptoVerifier.
+func NewLocalCryptoVerifier(cr crypto.Crypto, localKMS *localkms.LocalKMS) *LocalCryptoVerifier {
+	return &LocalCryptoVerifier{
+		cr:       cr,
+		localKMS: localKMS,
+	}
+}
+
+// Verify will verify a signature.
+func (v *LocalCryptoVerifier) Verify(pubKeyValue interface{}, doc, signature []byte) error {
+	var keyHandle *keyset.Handle
+
+	switch pubKey := pubKeyValue.(type) {
+	case *keyset.Handle:
```

Comment:
I don't really understand this change. 
It seems like we are using an interface{} and then requiring deep knowledge of the types (coupling the kms types).

ed25519 package should not know about tink.

Reply:
this is because `publicKeyValue` can be either a key reference (Tink's *keyset.Handle) or the raw []byte publicKey, similar to Crypto API that takes a `kh interface` argument for the key reference and the default implementation uses Tink's *keyset.Handle as kh type